### PR TITLE
Fixes #73 Remove extra characters in  last address when requesting 12 addresses

### LINKED ISSUE #73: Extra characters at the end of the last addess when requesting 12 addresses
**Describe the bug**
When requesting 12 addresses using the sample seed of the JS library (cloud flower upset remain green metal below cup stem infant art thank) the las address returned by the library is `ouE9prS4XMhVwQa5gwW4hQuiAGam\u0000\u0000\u0000\u0000\u0000\u0000`

A similar error happens when requesting the 50 first addresses.

This does not happens using the Go cli, so the problem must be on this library.

**Environment information:**

- OS: Windows 10
- Platform: Windows 10 pro x64
- Firmware Version: not sure

**Steps to Reproduce**
Call `deviceWallet.devAddressGen(12, 0, false, pinCodeReader, wordReader);` and see the response.

**Actual behavior**
The last address has some extra `\u0000` at the end.

**Expected behavior**
The address should not have extra characters.

**Additional context**
May be this is not too important as the hardware wallets will be limited to one address, at least for some time, and this library should be replaced with the daemon, but could be important in the future.

**Possible implementation**
Some time ago there was a similar problem with the function for signing transactions, but I'm not sure if the cause was the same. I think it was solved with https://github.com/skycoin/hardware-wallet-js/pull/35/commits/dc65883ee6ce115b4e9af5e24a1946660a794063

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #73 

Changes:
- Re-implemented the function `decodeAddressGenAnswer` to be able to validate the `null` sent by the HW

Does this change need to mentioned in CHANGELOG.md?
no

Requires changes in protobuff specs?
no

Requires testing
yes

Comments about testing , should you have some

### CODE REVIEW

File: device-wallet.js [hunk 1]
```diff
@@ -554,12 +555,36 @@ const decodeSignMessageAnswer =
...
+              // eslint-disable-next-line max-depth
+              for (let i = 2; i <= len + 1; i) {
+                // eslint-disable-next-line max-depth
+                if(dataBuffer[m + i] !== 0) {
+                  addr += String.fromCharCode(dataBuffer[m + i]);
+                }
+                i += 1;
+              }
+
+              addresses.push(addr);
```
Comment (olemis) [inline device-wallet.js]: if modified here can't be declared as `const` .

File: device-wallet.js [hunk 2]
```diff
@@ -554,12 +555,36 @@ const decodeSignMessageAnswer =
...
     function(kind, dataBuffer) {
-      let addresses = [];
-      if (kind == messages.MessageType.MessageType_ResponseSkycoinAddress) {
+      const addresses = [];
+      if (kind === messages.MessageType.MessageType_ResponseSkycoinAddress) {
         try {
-          const answer = messages.ResponseSkycoinAddress.decode(dataBuffer);
-          console.log('Addresses', answer.addresses);
-          addresses = answer.addresses;
+          for (let addr = "", findAddr = false, len, m = 0; m <= dataBuffer.length; m += 1 ) {
```
Comment (olemis) [inline device-wallet.js]: `m += 1` => `++m`

File: test/test_codes.js [hunk 3]
```diff
@@ -93,10 +93,10 @@ if (testSign) {
   );
 }
 
-const testAddressGen = false;
+const testAddressGen = true;
 
 if (testAddressGen) {
-  const promise = deviceWallet.devAddressGen(9, 3, true, pinCodeReader, wordReader);
+  const promise = deviceWallet.devAddressGen(12, 0, false, pinCodeReader, wordReader);
```
Comment (olemis) [inline test/test_codes.js]: why ?

Comment (stdevHan) [reply test/test_codes.js]: To be able to prove that what is asked in the issue works

File: device-wallet.js (hunk 2 above)
Comment (stdevHan) [reply device-wallet.js]: in JS, eslint defines this as an error, the correct way is so, so that at the time of npm run lint you do not get error

Comment (stdevIvr) [reply device-wallet.js]: For some reason Douglas Crockfor recommends to use `m += 1` instead `m++` or `++m`. Maybe is for ambiguity with iterators and pointer operations etc...

File: device-wallet.js (hunk 1 above)
Comment (stdevIvr) [reply device-wallet.js]: Addresses is not const, answer is.

File: device-wallet.js (hunk 2 above)
Comment (stdevIvr) [reply device-wallet.js]: But... it should be `m < dataBuffer.length`

File: test/genAddress-test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    deviceWallet.setAutoPressButton(true, 'R');
+  } else {
+    console.log("Skycoin hardware is plugged in");
+    deviceWallet.setDeviceType(deviceWallet.DeviceTypeEnum.USB);
+  }
+
+  it('Should match with known addresses', function() {
+    this.timeout(0);
+    return setup().
+      then(() => deviceWallet.devAddressGen(12, 0, false, pinCodeReader, wordReader)).
```
Comment (olemis) [inline test/genAddress-test.js]: Gotta set mnemonic before generating addresses

Comment (stdevIvr) [reply test/genAddress-test.js]: Yes, and that's a request for the test to run. If you don't set the mnemonic first, you get this error:

```bash
Failure message code 13 message:  Mnemonic required
```

Comment (olemis) [reply test/genAddress-test.js]: I do not see that failure in the [build logs](https://travis-ci.com/skycoin/hardware-wallet-js/builds/110472101) , and no mnemonic set in test code . Why ?

Comment (olemis) [reply test/genAddress-test.js]: @stdevStark I tested what @stdevIvr said and yes, mnemonic check is enforced . Could you take a look into this to see why test case not failing ?

Comment (stdevIvr) [reply test/genAddress-test.js]: Fixed that. You can select whether the mnemonic is set or not.

File: test/applySettings-test.js [hunk 5]
```diff
@@ -35,31 +35,13 @@ describe('Apply Setting -> label', function () {
...
-          return Promise.reject(new Error(`Unexpected failure message ${err.toString()}`));
-        }
-      );
+      then(() => done(new Error("Expected failure"))).
+      catch(() => {
+        done();
+      });
   });
 
-  it("Should not accept invalid languages", function() {
```
Comment (olemis) [inline test/applySettings-test.js]: restore this

**Pull request merged**